### PR TITLE
Change CompositeFileChangeToken to CompositeChangeToken

### PR DESCRIPTION
**Pull request by jbagga**
Addresses aspnet/Common [issue #163](https://github.com/aspnet/common/issues/163)
PR (https://github.com/aspnet/Common/pull/175)
@pranavkm

### CODE REVIEW

File: src/Microsoft.Extensions.FileProviders.Composite/CompositeFileProvider.cs [hunk 1]
```diff
@@ -93,8 +93,8 @@ public IChangeToken Watch(string pattern)
             {
                 return NullChangeToken.Singleton;
             }
-            var CompositeFileChangeToken = new CompositeFileChangeToken(changeTokens);
-            return CompositeFileChangeToken;
+            var CompositeChangeToken = new CompositeChangeToken(changeTokens);
```
Comment (natemcmaster) [inline src/Microsoft.Extensions.FileProviders.Composite/CompositeFileProvider.cs]: Nit: reduce to one line `return new CompositeChangeToken(changeTokens);`

File: test/Microsoft.Extensions.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs [hunk 2]
```diff
@@ -10,6 +10,7 @@
 using Microsoft.Extensions.FileProviders.Physical;
 using Microsoft.Extensions.FileSystemGlobbing.Tests.TestUtility;
 using Xunit;
+using Microsoft.Extensions.Primitives;
```
Comment (natemcmaster) [inline test/Microsoft.Extensions.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs]: Nit: sort usings.

**Comment by pranavkm** [comment]
⌚️ for @natemcmaster 's changes

**Comment by jbagga** [comment]
@pranavkm @natemcmaster Updated

**Comment by pranavkm** [comment]
:shipit: once we have a build with the changes in Common.

File: test/Microsoft.Extensions.FileProviders.Physical.Tests/Microsoft.Extensions.FileProviders.Physical.Tests.csproj [hunk 3]
```diff
@@ -32,17 +26,19 @@
...
     <PackageReference Include="xunit" Version="2.2.0-beta4-build3444" />
   </ItemGroup>
-
   <ItemGroup Condition=" '$(TargetFramework)' == 'netcoreapp1.1' ">
     <PackageReference Include="Microsoft.NETCore.App" Version="1.2.0-*" />
   </ItemGroup>
-
-</Project>
+  <ItemGroup>
+    <Compile Update="MockFileSystemWatcher.cs">
```
Comment (pranavkm) [inline test/Microsoft.Extensions.FileProviders.Physical.Tests/Microsoft.Extensions.FileProviders.Physical.Tests.csproj]: Can you remove this `ItemGroup`

File: test/Microsoft.Extensions.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs [hunk 4]
```diff
@@ -8,8 +8,8 @@
 using System.Threading.Tasks;
 using Microsoft.Extensions.FileProviders.Internal;
 using Microsoft.Extensions.FileProviders.Physical;
-using Microsoft.Extensions.FileSystemGlobbing.Tests.TestUtility;
 using Microsoft.AspNetCore.Testing;
```
Comment (pranavkm) [inline test/Microsoft.Extensions.FileProviders.Physical.Tests/PhysicalFileProviderTests.cs]: Not your fault, but sort usings.

File: shared/CompositeFileChangeToken.cs [hunk 5]
```diff
@@ -1,74 +0,0 @@
...
-
-        public IDisposable RegisterChangeCallback(Action<object> callback, object state)
-        {
-            var disposables = new List<IDisposable>(ChangeTokens.Count);
-            for (var i = 0; i < ChangeTokens.Count; i++)
-            {
-                var disposable = ChangeTokens[i].RegisterChangeCallback(callback, state);
-                disposables.Add(disposable);
-            }
-            return new CompositeDisposable(disposables);
```
Comment (pranavkm) [inline shared/CompositeFileChangeToken.cs]: Also delete `CompositeDisposable`?

File: test/Microsoft.Extensions.FileProviders.Physical.Tests/Microsoft.Extensions.FileProviders.Physical.Tests.csproj [hunk 6]
```diff
@@ -1,17 +1,12 @@
 ﻿<Project Sdk="Microsoft.NET.Sdk" ToolsVersion="15.0">
-
```
Comment (natemcmaster) [inline test/Microsoft.Extensions.FileProviders.Physical.Tests/Microsoft.Extensions.FileProviders.Physical.Tests.csproj]: I rather like these new-lines: keeps the file organized in logical chucks. I realize VS 2017 deleted these automatically. Can you reverse this change?

**Comment by natemcmaster** [comment]
Changes are fine. Just had one nit. :shipit: when @pranavkm approves too.

File: test/Microsoft.Extensions.FileProviders.Composite.Tests/CompositeFileProviderTests.cs [hunk 7]
```diff
@@ -243,64 +243,17 @@ public void Watch_CompositeChangeToken_RegisterChangeCallbackCorrectlyTransmitsA
             var hasBeenCalled = false;
             object result = null;
             object state = new object();
-            changeToken.RegisterChangeCallback(item =>
+            var registeredState = changeToken.RegisterChangeCallback(item =>
```
Comment (pranavkm) [inline test/Microsoft.Extensions.FileProviders.Composite.Tests/CompositeFileProviderTests.cs]: Do you use this variable?

**Pull request merged**